kroxylicious-kms-provider-hashicorp-vault/src/test/java/io/kroxylicious/kms/provider/hashicorp/vault/VaultKmsTlsIT.java

```diff
@@ -0,0 +1,188 @@
+ * API calls with TLS enabled on the server.
+ */
+class VaultKmsTlsIT {
+
+    private static final String VAULT_TOKEN = "token";
+
+    private static final String HASHICORP_VAULT = "hashicorp/vault:1.15";
+    public static final int TLS_PORT = 8202;
+    @SuppressWarnings("rawtypes")
+    private static VaultContainer vaultContainer;
+    private static final Map<String, VaultKms> vaultKmsMap = new HashMap<>();
+    private static URI vaultTlsUri;
+
+    private static final CertificateGenerator.Keys keys = CertificateGenerator.generate();
+
+    static List<String> vaultKms() {
+        return vaultKmsMap.keySet().stream().toList();
+    }
+
+    @BeforeAll
+    public static void beforeAll() {
+        assumeThat(DockerClientFactory.instance().isDockerAvailable()).withFailMessage("docker unavailable").isTrue();
+        startVault();
+        String tlsAddress = String.format("https://%s:%s", vaultContainer.getHost(), vaultContainer.getMappedPort(TLS_PORT));
+        vaultTlsUri = URI.create(tlsAddress);
+        vaultKmsMap.put("pkcs12Tls", getTlsVaultKms(tlsForTrustStoreInlinePassword(keys.pkcs12ClientTruststore())));
+        vaultKmsMap.put("pkcs12NoPasswordTlsService", getTlsVaultKms(tlsForTrustStoreNoPassword(keys.pkcs12NoPasswordClientTruststore())));
+        vaultKmsMap.put("filePasswordTls", getTlsVaultKms(tlsForTrustStoreFilePassword(keys.pkcs12ClientTruststore())));
+        vaultKmsMap.put("jksTls", getTlsVaultKms(tlsForTrustStoreInlinePassword(keys.jksClientTruststore())));
+        vaultKmsMap.put("defaultStoreTypeTls", getTlsVaultKms(defaultStoreTypeTls(keys.jksClientTruststore())));
+        vaultKmsMap.put("tlsInsecure", getTlsVaultKms(insecureTls()));
+    }
+
+    private static void startVault() {
+        vaultContainer = new VaultContainer<>(HASHICORP_VAULT)
+                .withVaultToken(VAULT_TOKEN)
+                .withEnv("VAULT_FORMAT", "json")
+                // in vault server -dev mode the 8200 port cannot be reconfigured to TLS and is also useful so that the
+                // internal vault commands we make while execed into the container can communicate with vault over plain
+                // without having to trust the TLS cert within the container. We do not expose the plain listener to
+                // prevent false positives if the test connected to the wrong endpoint.
+                .withExposedPorts(TLS_PORT)
+                .withCopyFileToContainer(MountableFile.forClasspathResource("Vault.hcl"), "/vault/config/Vault.hcl")
+                .withCopyFileToContainer(MountableFile.forHostPath(keys.serverCertPem()), "/vault/config/cert.pem")
+                .withCopyFileToContainer(MountableFile.forHostPath(keys.serverPrivateKeyPem()), "/vault/config/key.pem")
+                .withInitCommand(
+                        "secrets enable transit");
+        vaultContainer.setWaitStrategy(Wait.forHttp("/v1/sys/health").forPort(TLS_PORT).usingTls().allowInsecure().forStatusCode(200));
```

Comment:
you add a wait strategy here, but not to the non-TLS test case.  Is that intentional?

Reply:
Yes, the default needs to be overridden for our TLS usage. In the TLS test we only expose the TLS port (to prevent any chance of accidentally connecting to the plain port) and the testcontainers wait code fails to connect because it doesn't trust the cert.